src/stirling/source_connectors/socket_tracer/protocols/common/lazy_contiguous_data_stream_buffer_impl.cc

```diff
@@ -151,9 +151,19 @@ void LazyContiguousDataStreamBufferImpl::MergeContiguousEventsIntoHead() {
   }
 
   it = events_.begin();
+  // end_it stopped at the first non-contiguous event (at the end of current head)
```

Comment:
Are we guaranteed to call `MergeContiguousEventsIntoHead` for the situations we care about? I noticed that `Head` only calls this function if `IsHeadAndEventsMergeable` is true ([source](https://github.com/pixie-io/pixie/blob/62b8080699371b0bcd83e2c3ddda265d22d0eb40/src/stirling/source_connectors/socket_tracer/protocols/common/lazy_contiguous_data_stream_buffer_impl.cc#L169)).

Reply:
`IsHeadAndEventsMergeable` is true whenever we have events that line up with the end of the previous `head_` such that we can merge at least one event from `events_`.

Since we only check for monotonicity when merging events into `head_`, any existing `head_` will have monotonic timestamps.

Does that answer your question?

Reply:
Yea, just wanted to double check my understanding of that merge logic 👍